config/custom.go

```diff
@@ -543,6 +543,52 @@ func decodeStringToMapHookFunc() mapstructure.DecodeHookFunc {
 	})
 }
 
+var (
+	ErrRuntimeFlagUnknown         = fmt.Errorf("decoding runtime flags: unknown flag")
```

Comment:
Is there a good reason to enforce this? Doesn't this just limit the flexibility of runtime flags? (e.g. we remove one in the future, and now all config that had it fails, we add one an older version of pomerium fails because it doesn't exist there, etc...)

Reply:
I believe runtime flags are not covered by any version compatibility contract. 

I don't expect users to flip those flags unless explicitly instructed. 

It's like a command line flags - if you provide a flag to a program version that doesn't support it the customary behaviour is to quit with error, not to ignore. 

Note this only applies to flags supplied via config file and not via the protobuf.

Reply:
My understanding is that currently our default is to warn on unknown config fields. I can add a follow up issue to error out.

This will bite us, but ok.

Reply:
we can keep current warning behaviour for consistency.